### DISCUSSION TITLE
Can Empty Blocks Be Added To Create Form?

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#8407] [General]
**I need to know if its possible to pre-populate builder blocks on create form without clicking the action label.**

![screenshot](https://github.com/filamentphp/filament/assets/107990171/0fc54601-90ed-4998-97d3-8d6558770692)

**Comment by leandrocfe** [comment]
```php
Builder::make('content')
    ->blocks([
        Builder\Block::make('heading')
            ->schema([
                TextInput::make('content'),
            ]),
        // ...
    ])
    ->default([
        [
            'type' => 'heading',
            'data' => [
                'content' => 'Value 1'
            ]
        ],
        [
            'type' => 'heading',
            'data' => [
                'content' => 'Value 2'
            ]
        ],
    ])
```

**Reply by yuvrajtimalsina** [reply]
I wanted to pre-populate empty blocks without me adding them in create form. For example if I have a block with paragraph setup in my form, I want it to be open while I am trying to create a new model record.

**Comment by leandrocfe** [comment]
`TextInput::make('content')->default('xx')`
?

**Reply by yuvrajtimalsina** [reply]
Setting a default value didn't trigger my block fields to be open and active while creating a record.

**Reply by leandrocfe** [reply]
Sorry, I didn't understand what you need

**Reply by yuvrajtimalsina** [reply]
Concept of Builder Blocks is to add predefined blocks after we click the button and choose the block we want to add.

But, if I have two blocks : TextInput block and FileUpload block. Then,I want form field to display TextInput without me clicking the button to select a block by default at first if possible.

**Reply by leandrocfe** [reply]
Did you try my previous code in the create page?

**Reply by yuvrajtimalsina** [reply]
Yes, I did. Nothing happened there and I had to manually add the Block.

**Reply by leandrocfe** [reply]
This was supposed to work
```php
Builder::make('content')
    ->blocks([
        Builder\Block::make('heading')
            ->schema([
                TextInput::make('content'),
            ]),
        // ...
    ])
    ->default([
        [
            'type' => 'heading',
            'data' => [
                'content' => 'Value 1'
            ]
        ],
        [
            'type' => 'heading',
            'data' => [
                'content' => 'Value 2'
            ]
        ],
    ])
```

![image](https://github.com/filamentphp/filament/assets/3833889/889dc5ce-d24c-4b14-9df8-e40a0f0ef6c0)

**Reply by yuvrajtimalsina** [reply]
Were the fields auto opened while you entered create page because it didn't happen in my case.

**Reply by leandrocfe** [reply]
yep, share your code..

**Comment by yuvrajtimalsina** [comment]
My form was too large and I misplaced some blocks and to my surprise no error occured.

It worked now, thank you for the response brother. 🫡

**Comment by aymanalhattami** [comment]
Does anyone know a way to generate the builder blocks based on state changes?
